hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineCache.java

```diff
@@ -52,12 +52,13 @@ public AlertsEngineCache() {
     /**
      * Check if a specific dataId is active on this node
      *
+     * @param tenantId to check if has triggers deployed on this node
      * @param dataId to check if it has triggers deployed on this node
      * @return true if it is active
      *         false otherwise
      */
-    public boolean isDataIdActive(String dataId) {
-        return dataId != null && activeDataIds.contains(dataId);
+    public boolean isDataIdActive(String tenantId, String dataId) {
+        return tenantId != null && dataId != null && activeDataIds.contains(new DataId(tenantId, dataId));
```

Comment:
OK, so if in a distributed environment triggers with the same ids and different tenants can be processed on different nodes, then this makes sense.